protos/flyteidl/core/tasks.proto

```diff
@@ -194,6 +194,25 @@ message DataLoadingConfig {
   // In the inputs folder, there will be an additional summary/metadata file that contains references to all files or inlined primitive values.
   // This format decides the actual encoding for the data. Refer to the encoding to understand the specifics of the contents and the encoding
   MetadataFormat format = 3;
-   // Flag enables DataLoading Config. If this is not set, data loading will not be used! 
-   bool enabled = 4;
+  // Flag enables DataLoading Config. If this is not set, data loading will not be used! 
+  bool enabled = 4;
+  // Strategy to use when dealing with Blob, Schema, or multipart blob data (large datasets)
+  enum BlobDownload {
+    // All data will be downloaded before the main container is executed
+    BEFORE_STARTUP = 0;
+    // Data will be downloaded as a stream and an End-Of-Stream marker will be written to indicate all data has been downloaded. Refer to protocol for details
+    STREAM = 1;
+    // Large objects (offloaded) will not be downloaded
+    DO_NOT_DOWNLOAD = 2;
+  }
```

Comment:
These names will get all mangled up, right? 
They will look something like DataLoadingConfig_BEFORE_STARTUP, DataLoadingConfig_ON_EXIT... at least in golang, you won't know which enum they belong to...
You might want to create a nested message, something like this:
```suggestion
  message BlobDownload {
      enum Strategy {
          BEFORE_STARTUP = 0;
          ...
      }
 }
 
 BlobDownload.Staregy blob_download_strategy = 5;
```

Reply:
ya they get mangled up. But creating a message is an overhead? Maybe i can create a message called 
message IOStrategy{
enum Download
enum Upload
}